### PR TITLE
Add byte array to numeric new functions + UINT256 support

### PR DESCRIPTION
**Pull request by belen-pruvost**

### CODE REVIEW

**Comment by belen-pruvost** [comment]
@jkylling do you think I should keep the link to trino docs on numerical types? And I wasn't sure on whether to add the functions we support for the uint256.

File: docs/reference/dune-v2/query-engine.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ The syntax and keyword operator differences between Postgres, Spark, and Dune SQ
 
 | <div style="width:290px">**Description**</div> | **V1 - PostgreSQL** | **V2 - Spark SQL** | **V2 - Dune SQL** |
 | --- | --- | --- | --- |
-| **`bytea2numeric` does not exist in Spark.** | `bytea2numeric` (bytea) | `bytea2numeric_v2` (string) | `bytea2numeric` (string) |
+| **`bytea2numeric` does not exist in Spark.** | `bytea2numeric` (bytea) | `bytea2numeric_v2` (string) | `bytearray_to_integer` (hex) <br> `bytearray_to_bigint` (hex) <br> `bytearray_to_decimal` (hex) <br> `bytearray_to_uint256` (hex) <br> `bytea2numeric` (hex) <br> More details on [Byte Array to Numeric Functions](#byte-array-to-numeric-functions)|
```
Comment (jkylling) [inline docs/reference/dune-v2/query-engine.md:24]: Add deprecation notice on bytearray2numeric

Comment (jkylling) [reply docs/reference/dune-v2/query-engine.md:24]: Or omit it

File: docs/reference/dune-v2/query-engine.md [hunk 2]
```diff
@@ -50,6 +50,21 @@ This is because the parser sometimes treats words in double quotes as a string a
...
+
+## Byte Array to Numeric Functions
+- `bytearray_to_integer`, returns the `INTEGER` value of a big-endian byte array of length <= 4 representing the integer in two's complement. If the byte array has length < 4 it is padded with zero bytes.
+- `bytearray_to_bigint`, returns the `BIGINT` value of a big-endian byte array of length <= 8 representing the bigint in two's complement. If the byte array has length < 8 it is padded with zero bytes.
+- `bytearray_to_decimal`, returns the `DECIMAL(38,0)` value of a big-endian byte array of length <= 16 representing the decimal(38,0) in two's complement. If the byte array has length < 16 it is padded with zero bytes.
+- `bytearray_to_uint256`, returns the `UINT256` of a big-endian byte array of length <= 32 representing the unsigned integer. If the byte array has length < 32 it is padded with zero bytes.
+- `bytea2numeric` behaves exactly as `bytearrat_to_bigint`
```
Comment (jkylling) [inline docs/reference/dune-v2/query-engine.md:61]: Add deprecation notice. Rewrite:
`bytea2numeric` is deprecated. It is an alias for `bytearray_to_bigint`.

File: docs/reference/dune-v2/query-engine.md [hunk 3]
```diff
@@ -50,6 +50,21 @@ This is because the parser sometimes treats words in double quotes as a string a
...
+
+The functions throw an overflow exception if the byte array is larger than the number of bytes supported for the type, even if the most significant bytes are all zero. For this cases, it is possible to use `bytearray_ltrim` in order to trim the left zero bytes.
```
Comment (jkylling) [inline docs/reference/dune-v2/query-engine.md:63]:
```suggestion
The functions throw an overflow exception if the byte array is larger than the number of bytes of the type, even if the most significant bytes are all zero. It is possible to use `bytearray_ltrim` to trim the zero bytes from the left.
```

Comment (jkylling) [inline docs/reference/dune-v2/query-engine.md:63]:
```suggestion
The bytearray conversion functions throw an overflow exception if the byte array is larger than the number of bytes supported for the type, even if the most significant bytes are all zero. For this cases, it is possible to use `bytearray_ltrim` in order to trim the left zero bytes.
```

File: docs/reference/dune-v2/query-engine.md [hunk 4]
```diff
@@ -50,6 +50,21 @@ This is because the parser sometimes treats words in double quotes as a string a
 
 For example, referencing a column name in the `WHERE` clause using double quotes works as expected. However, the same query inside a CTE treats the column name as a string, [as can be seen here](https://dune.com/queries/1199604).
 
+## Numerical types
+We support the [numerical types](https://trino.io/docs/current/language/types.html) `INTEGER`, `BIGINT`, `DOUBLE`, and fixed precision `DECIMAL` with precision up to 38 digits (i..e, `DECIMAL(38, 0)`). Additionally, we have `UINT256` for doing arithmetic on unsigned 256 bit integers.
```
Comment (jkylling) [inline docs/reference/dune-v2/query-engine.md:54]:
```suggestion
We support the [numerical types](https://trino.io/docs/current/language/types.html) `INTEGER`, `BIGINT`, `DOUBLE`, and fixed precision `DECIMAL` with precision up to 38 digits (i..e, `DECIMAL(38, 0)`). Additionally, we support `UINT256` for representing unsigned 256 bit integers.
```

**Review by jkylling — APPROVED** [review]
Just some suggested rewritings. Feel free to ignore.

**Pull request merged**